src/v/cluster/producer_state_manager.h

```diff
@@ -35,9 +35,10 @@ class producer_state_manager
     void deregister_producer(producer_state&);
     // temporary relink already accounted producer
     void link(producer_state&);
+    void rearm_timer_for_testing(std::chrono::milliseconds);
 
 private:
-    static constexpr std::chrono::seconds period{5};
+    std::chrono::milliseconds period{5000};
```

Comment:
nit: since this is now a private variable please prefix it with `_` for consistency, maybe a name like `_reaper_period` would be more appropriate ?